Src/Optimization/ISolver.h

```diff
@@ -3,9 +3,17 @@
 #include <vector>
 
 #include "IFunction.h"
+#include "Test.h"
 
 class ISolver
 {
 public:
-    virtual double solve(const IFunction& f, double a, double b, double epsilon, int n) = 0;
+    virtual void performTest(IFunction& f, double a, double b, double epsilon, int n) = 0;
+    virtual void evaluateSolution() = 0;
+    virtual bool checkStopCriterion(double epsilon, int n) = 0;
+    virtual double solve(IFunction& f, double a, double b, double epsilon, int n, std::vector<Test>& history, int& operationCount) = 0;
```

Comment:
`std::vector<Test>& history, int& operationCount` what as a user should I pass  here? Why this variables in the public interfaec?